src/modules/keyboardmanager/ui/ShortcutControl.cpp

```diff
@@ -81,6 +85,14 @@ void ShortcutControl::AddNewShortcutControlRow(Grid& parent, std::vector<std::ve
     targetAppTextBox.PlaceholderText(KeyboardManagerConstants::DefaultAppName);
     targetAppTextBox.Text(targetAppName);
 
+    // To set the accessible name of the target App text box by adding the string `All Apps` if the text box is empty, if not the application name is read by narrator.
+    std::wstring targetAppTextBoxAccessibleName = GET_RESOURCE_STRING(IDS_TARGET_APPLICATION);
+    if (targetAppTextBox.Text() == L"")
+    {
+        targetAppTextBoxAccessibleName += GET_RESOURCE_STRING(IDS_EDITSHORTCUTS_ALLAPPS);
+    }
+    targetAppTextBox.SetValue(Automation::AutomationProperties::NameProperty(), box_value(targetAppTextBoxAccessibleName));

```

Comment:
If we are adding in logic for this, then we should consider adding these lines at the end of the LostFocus handler, otherwise there could be situations where it reads out All Apps and some app name.

Reply:
By having the lines in both places, the correct name would be set on initialization and whenever any change is made in the textbox. Since it would have to be duplicated maybe it could be added as a local function in this file to reduce code duplication.

Reply:
Updated it.